src/main/java/uk/gov/hmcts/reform/sendletter/services/ReportsService.java

```diff
@@ -1,16 +1,72 @@
 package uk.gov.hmcts.reform.sendletter.services;
 
-import org.apache.commons.lang.NotImplementedException;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sendletter.entity.LettersCountSummaryRepository;
+import uk.gov.hmcts.reform.sendletter.entity.reports.ServiceLettersCountSummary;
 import uk.gov.hmcts.reform.sendletter.model.out.LettersCountSummary;
+import uk.gov.hmcts.reform.sendletter.services.ftp.ServiceFolderMapping;
 
 import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.reform.sendletter.util.TimeZones.EUROPE_LONDON;
+
 @Service
 public class ReportsService {
 
+    private static final String TEST_SERVICE = "BULKPRINT";
+
+    private final LettersCountSummaryRepository repo;
+
+    private final ServiceFolderMapping serviceFolderMapping;
+
+    private final ZeroRowFiller zeroRowFiller;
+
+    private final String downtimeFromHour;
+
+    private final String downtimeToHour;
+
+    public ReportsService(
+        LettersCountSummaryRepository repo,
+        ServiceFolderMapping serviceFolderMapping,
+        ZeroRowFiller zeroRowFiller,
+        @Value("${ftp.downtime.from}") String downtimeFromHour,
+        @Value("${ftp.downtime.to}") String downtimeToHour
+    ) {
+        this.repo = repo;
+        this.serviceFolderMapping = serviceFolderMapping;
+        this.zeroRowFiller = zeroRowFiller;
+        this.downtimeFromHour = downtimeFromHour;
+        this.downtimeToHour = downtimeToHour;
+    }
+
     public List<LettersCountSummary> getCountFor(LocalDate date) {
-        throw new NotImplementedException("Not yet implemented");
+        LocalDateTime dateTimeFrom = formatDateTime(date.minusDays(1), LocalTime.parse(downtimeFromHour));
+        LocalDateTime dateTimeTo = formatDateTime(date, LocalTime.parse(downtimeToHour));
+
+        return zeroRowFiller.fill(
+            repo.countByDate(dateTimeFrom, dateTimeTo).stream().map(this::fromDb).collect(toList()))
+            .stream()
+            .filter(summary -> !summary.service.equals(TEST_SERVICE))
+            .collect(toList()); //exclude test service
+    }
+
+    private LettersCountSummary fromDb(ServiceLettersCountSummary dbSummary) {
+        return new LettersCountSummary(
+            serviceFolderMapping.getFolderFor(dbSummary.getService()).orElse(null),
```

Comment:
should be service name, not folder

Reply:
we will have `divorce_frontend` if we use service name and I am sure we would expect to see `DIVORCE` rather then `divorce_frontend`

Reply:
the service names are "cmc_claim_store", "divorce_frontend", "probate_backend" and "sscs". 
The folder names would be clearer on the report which would be CMC, DIVORCE, PROBATE and SSCS

Reply:
good thing it's not `TO_XEROX` anymore

Reply:
I still think it would be more sensible to use service name instead of name of a folder on some FTP server.
Folder names just happen to looks like something that business would like.
If service names are not clear (to be checked) we could add a mapping between actual service name and service reports display name.

Reply:
we had a team discussion and decided to create a ticket to add service name property to the mappings config.